server/core/src/logic/expense_operations.rs

```diff
@@ -1,34 +1,44 @@
 use self::errors::{
-    CreateExpenseError, CreatePredefinedExpenseError, FindExpensesByUserIdError,
+    CreateExpenseError, CreatePredefinedExpenseError, FindExpensesWithTransactionsByUserIdError,
     ValidateRecurrenceAndCurrencyTypesError,
 };
 
 use super::common;
 use super::user_operations::authorize_user_by_id;
-use crate::dto::expenses::{NewExpenseRequest, NewPredefinedExpenseRequest};
+use crate::dto::expenses::{
+    ExpenseWithTransactions, NewExpenseRequest, NewPredefinedExpenseRequest,
+};
 use crate::logic::currency_operations::find_currency_type_by_id;
 use crate::logic::recurrence_operations::find_recurrence_type_by_id;
 
 use entity::expense::{self, Entity as Expense};
 use entity::predefined_expense::{self, Entity as PredefinedExpense};
+use entity::transaction::Entity as Transaction;
 use entity::Id;
 
 use chrono::NaiveDate;
 use migration::DbErr;
 use sea_orm::ActiveValue::NotSet;
 use sea_orm::{ActiveModelTrait, ColumnTrait, DatabaseConnection, EntityTrait, QueryFilter, Set};
 
-pub async fn find_expenses_by_user_id(
+pub async fn find_expenses_with_transactions_by_user_id(
     conn: &DatabaseConnection,
     authenticated_user_id: Id,
     user_id: Id,
-) -> Result<Vec<expense::Model>, FindExpensesByUserIdError> {
+) -> Result<Vec<ExpenseWithTransactions>, FindExpensesWithTransactionsByUserIdError> {
     authorize_user_by_id(authenticated_user_id, user_id)?;
-    let expenses = Expense::find()
+    let mut result: Vec<ExpenseWithTransactions> = vec![];
+    let expenses_with_transactions = Expense::find()
+        .find_with_related(Transaction)
         .filter(expense::Column::UserId.eq(user_id))
         .all(conn)
         .await?;
-    Ok(expenses)
+
+    for expense in expenses_with_transactions {
```

Comment:
I would like to see a more functional approach here. Also I think you can destructure expense like so `for (expense, transactions) in expenses_with_transactions ...` and the tuple numbering would be more readable.

Reply:
Fixed